src/include/types.h

```diff
@@ -476,8 +476,10 @@ ostream &operator<<(ostream &lhs, const shard_id_t &rhs);
 
 #if defined(__sun) || defined(_AIX) || defined(DARWIN)
 __s32  ceph_to_host_errno(__s32 e);
+__s32  host_to_ceph_errno(__s32 e);
```

Comment:
@tchaikov 
I submitted this part separate on request from Sage.
I'm adding the other stuff to my original PR #15495.
Note, like you suggest, that the code to compile SUN|AIX|DARWIN eerno.cc is not in Cmake ATM.